tests/Kernel/ContentEventRenderTest.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\file\Entity\File;
+use Drupal\Tests\rdf_entity\Traits\RdfDatabaseConnectionTrait;
+use Drupal\user\Entity\Role;
+use Drupal\user\RoleInterface;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests that our Event content type render.
+ *
+ * @todo: Extend this test with more scenarios.
+ */
+class ContentEventRenderTest extends AbstractKernelTestBase {
```

Comment:
Cant we extend from `ContentRenderTest ` ?

Reply:
Because here we don't need news, pages, publication and policy content types to be install, but many other for event content type only. Therefore the parent::setup method can't be executed but the one in one more level down, the AbstractKernelTest.

Reply:
Can't we turn this into a browser test? I've tried it locally and it sets up in 85 seconds. 

![image](https://user-images.githubusercontent.com/153362/86591112-33e48980-bf91-11ea-80d6-5850540f92b1.png)

All you need to do is to replace the initial part of this test with the following:

```php
class ContentEventRenderTest extends BrowserTestBase {

  /**
   * {@inheritdoc}
   */
  public static $modules = [
    'config',
    'system',
    'oe_theme_helper',
    'path',
    'oe_theme_content_event',
    'content_translation',
  ];

  /**
   * {@inheritdoc}
   */
  protected function setUp() {
    parent::setUp();

    // Enable and set OpenEuropa Theme as default.
    $this->container->get('theme_installer')->install(['oe_theme']);

    // Make event node and image media translatable.
    $this->container->get('content_translation.manager')->setEnabled('node', 'oe_event', TRUE);
    $this->container->get('content_translation.manager')->setEnabled('media', 'image', TRUE);
    $this->container->get('router.builder')->rebuild();
  }
```

Reply:
Yes, but to debug this is a pure nightmare. It's double the time easily is counter-productive. The only way I could imagine to go for such robust tests is with a local class that skips the setup after the first run and restores a database dump from the point when the setup finished and before anything ran.

Reply:
Also, for single scenario it could be okay, but as soon as you need to check 4-5 content types and different test scenarios the time quickly increases from 84 seconds to 500 seconds impacting the project test time on drone.